buffer/src/main/java/io/netty/buffer/AbstractByteBufAllocator.java

```diff
@@ -50,6 +50,28 @@ protected static ByteBuf toLeakAwareBuffer(ByteBuf buf) {
         return buf;
     }
 
+    protected static CompositeByteBuf toLeakAwareBuffer(CompositeByteBuf buf) {
+        ResourceLeak leak;
+        switch (ResourceLeakDetector.getLevel()) {
```

Comment:
Can the level change? If not, would it be make sense to save off some sort of allocator for the leak aware buffers?